File: backend/src/main/java/com/etekhno/rentacar/domain/PendingPaymentDTO.java
package com.etekhno.rentacar.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class PendingPaymentDTO {
    String customerId;
    String customerCNIC;
    String customerName;
    BigDecimal remainingBalance;
}

File: frontend/lib/domains/pending_payment_dto.dart
class PendingPaymentDTO {
  String? id;
  String? customerId;
  String? customerName;
  double? remainingBalance;

  PendingPaymentDTO({
    this.id,
    this.customerId,
    this.customerName,
    this.remainingBalance,
  });

  factory PendingPaymentDTO.fromJson(Map<String, dynamic> json) {
    return PendingPaymentDTO(
      id: json['id'],
      customerId: json['customerId'],
      customerName: json['customerName'],
      remainingBalance: json['remainingBalance']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'customerId': customerId,
      'customerName': customerName,
      'remainingBalance': remainingBalance,
    };
  }
}